neuro_flow/batch_runner.py

```diff
@@ -192,10 +192,11 @@ async def process(
                 await asyncio.sleep(1)
 
     def _accumulate_result(self, finished: Iterable[FinishedTask]) -> JobStatus:
-        # TODO: handle cancelled tasks
         for task in finished:
             if task.status == JobStatus.FAILED:
                 return JobStatus.FAILED
+            elif task.status == JobStatus.CANCELLED:
+                return JobStatus.CANCELLED
```

Comment:
What should be returned if some tasks were failed and some tasks were cancelled?

Reply:
That's a good question, thank you!

My thoughts are as follows:
If the parent task was cancelled, all child tasks should be cancelled as well, no failed tasks should arise because of the cancelled parent. If the parent task failed, all child tasks should be automatically cancelled. However, it works only for a chain of tasks.

But or case is a DAG of tasks, where the things more complicated.
I assume, that in case of the cancelled task, it's depended tasks are not cancelled automatically but allowed to fail. @asvetlov, is it correct?
If so, I will gather all statuses, check whether we have `cancelled` status and report respectively.

However, if the depended tasks are also cancelled, a graph should be analysed in the branch-wise approach.

Reply:
It looks reasonable. Thanks.

Reply:
Would be nice to add a clarifying comment.

Reply:
@serhiy-storchaka, I'll add a clarifying comment.
@asvetlov, the question is still open: if some task in batch mode is cancelled, it's dependend tasks are cancelled authomatically, or they are allowed to fail?

Reply:
github has a set of [job status check functions](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions) with `success()` by default.
I think we need something similar.

Reply:
That would be a cool feature to have in neuro-flow, I totally agree. However, it is related to the control flow within a graph, not to the bake attempt results reporting, which was modified here.

Since we have a topologically sorted graph of tasks, I've changed the order of reporting status to `cancelled -> failed -> succeeded` in 02d1dbe. It seems to be more logical.

@asvetlov, maybe I've got your comment wrong, please, clarify. Otherwise, if all is OK, let's proceed with this PR.

Reply:
changes are approved during call